### ISSUE TITLE
Create OpenAPI Spec for Redesigned API

### ISSUE THREAD

**Issue by ahshabbir** [UTDNebula/nebula-api#196]
Since many of the consumer facing changes of the API Redesign have been finalized, we can edit the OpenAPI spec to reflect those changes and send them to our consumers right away while we discuss some implementation details. These are the changes that we should detail:

## Refactoring the grades endpoint

Currently, our grades endpoint has all the possibilities for a grades bundle you could recieve. For example, for a particular course:

`GET /grades?subject_prefix=CS&course_number=1337`

Instead, we should get rid of the grades endpoint in this form and instead frame it as attributes of a course, professor, section, etc.

`GET /course/{courseId}/grades`

Where courseId could be the ID of CS 1337.

## Request in the body or as query parameters

We've decided to keep the request as query parameters, so no changes are needed here.

Note that this isn't an implementation issue, this is just a documentation change in the OpenAPi Spec. Thus, #195 should be completed first.

**Comment by jpahm** [comment]
Moving grade endpoint refactor to its own issue and closing this, as we now have automated OpenAPI spec generation due to #215
